### DISCUSSION TITLE
Is toDouble missing?

### DISCUSSION THREAD

**Discussion by fischerscode** [ethanblake4/dart_eval#205] [Q&A]
Hi there,

I'm relying on AI to generate code that should then be evaluated, therefore I need `dart_eval` to be quite close to actual `dart`. The other day, I got an error that `toDouble` is not working. Back then, I had no really an Idea how `dart_eval` works, assumed that I'm doing something wrong, and just multiplied with `1.0` instead.

Since I hate writing boilerplate and my project requires Wrappers that are not supported by `bindgen`, I'm currently working on [my own wrapper builder](https://github.com/fischerscode/pub.packages/tree/eval_builder/packages/eval_builder). (Disclaimer: When I started the project, I missed that `bindgen` exists in the first place.)

Now I have a better understanding of how wrappers operate and stumbled over the definition of `$num` and `$int`. From what I can tell, they actually don't support most of the common dart features.

1. Am I missing something?
2. Or are actually just the (few) methods defined in the `BridgeClassDef` supported?
3. And where are the operators of `$num.$getProperty` defined?
4. And if something like `toDouble` is actually not supported, is it safe to `runtime.registerBridgeFunc` a custom `$int` wrapper? (Didn't even test if that works, but it seems like the later defined stuff [is prioritized](https://github.com/ethanblake4/dart_eval/blob/b06ed0de64bf1c494f07497fec1e4e8a59f02854/lib/src/eval/runtime/runtime.dart#L235-L251).)

My current workaround would be the following:
1. Make my builder quite versatile
2. Generate Wrappers for the needed dart types like `int`.
3. Try to override the wrappers that `dart_eval` uses.

By the way, `dart_eval` seems to be such an unique and powerful project. 👏 

Edit:
I just found [getKnownMethods](https://github.com/ethanblake4/dart_eval/blob/b06ed0de64bf1c494f07497fec1e4e8a59f02854/lib/src/eval/compiler/builtins.dart#L99C1-L99C12). Maybe I can inject into the map, or at least this seams to be a starting point for a PR.

**Comment by Zverik** [comment]
`toDouble` is present in 0.8.
